### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by AnnSerdechna**
[DEMO LINK](https://annserdechna.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

**Review by jursylian — APPROVED** [review]
Super! Good job!

File: src/App.tsx [hunk 1]
```diff
@@ -12,13 +18,107 @@ const goodsFromServer: string[] = [
...
-);
+].map(good => ({
+  name: good,
+  id: uuidv4(),
+}));
+
+const App: React.FC = () => {
+  const [goods, setGoods] = useState([...goodsFromServer]);
+  const [isListVisible, setIsListVisible] = useState(false);
+  const [value, setValue] = useState('1');
```
Comment (Ihor-Karpyn) [inline src/App.tsx:29]: not clear, the value of what?

File: src/App.tsx [hunk 2]
```diff
@@ -12,13 +18,107 @@ const goodsFromServer: string[] = [
...
+
+  const onChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
```
Comment (Ihor-Karpyn) [inline src/App.tsx:31]: onChange what?

File: src/App.tsx [hunk 3]
```diff
@@ -12,13 +18,107 @@ const goodsFromServer: string[] = [
...
+  };
+
+  const sortedByLength = () => {
+    const sorted = [...goods].sort((a, b) => a.name.length - b.name.length);
+
+    setGoods(sorted);
+  };
+
+  const showList = () => {
+    setIsListVisible(!isListVisible);
```
Comment (Ihor-Karpyn) [inline src/App.tsx:48]: Just `true`, if this function work like toggler it's should have name listVissibleToggler
```suggestion
    setIsListVisible(true);
```

File: src/App.tsx [hunk 4]
```diff
@@ -12,13 +18,107 @@ const goodsFromServer: string[] = [
...
+  };
+
+  const reverse = () => {
+    setGoods([...goods].reverse());
+  };
+
+  const reset = () => {
```
Comment (Ihor-Karpyn) [inline src/App.tsx:55]: btn reset should reset all filters

File: src/App.tsx [hunk 5]
```diff
@@ -12,13 +18,107 @@ const goodsFromServer: string[] = [
...
+    <div className="App">
+      <button
+        type="button"
+        onClick={showList}
+      >
+        {isListVisible ? 'Hide' : 'Start'}
+      </button>
+
+      {isListVisible
+        && (
```
Comment (Ihor-Karpyn) [inline src/App.tsx:69]:
```suggestion
      {isListVisible && (
```

File: src/App.tsx [hunk 6]
```diff
@@ -12,13 +18,107 @@ const goodsFromServer: string[] = [
...
+  const [value, setValue] = useState('1');
+
+  const onChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setValue(e.target.value);
+  };
+
+  const sortedByAlphbet = () => {
+    const sorted = [...goods].sort((a, b) => a.name.localeCompare(b.name));
+
+    setGoods(sorted);
```
Comment (Ihor-Karpyn) [inline src/App.tsx:38]: create var preparedTodos. Don't save sorted or reversed todos in the state

**Review by Ihor-Karpyn — CHANGES_REQUESTED** [review]

**Review by Ihor-Karpyn — APPROVED** [review]

**Pull request closed without merging**